### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: trie/trieBatchManager/trieBatchManager_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	assert.True(t, found)
+	_, found = batchManager.Get([]byte("key2"))
+	assert.True(t, found)
+
+	batchManager, err = tbm.MarkTrieUpdateInProgress()
+	assert.True(t, check.IfNil(batchManager))
+	assert.Equal(t, ErrTrieUpdateInProgress, err)
+
+	tbm.MarkTrieUpdateCompleted()
+	assert.False(t, tbm.isUpdateIsProgress)
```
Comment (AdoAdoAdo) [inline trie/trieBatchManager/trieBatchManager_test.go:45]: typo isUpdateIsProgress - isUpdateInProgress

File: trie/trieBatchManager/trieBatchManager_test.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		})
+		_, _ = tbm.MarkTrieUpdateInProgress()
+		_, found := tbm.currentBatch.Get(key)
+		assert.False(t, found)
+
+		data, found := tbm.Get(key)
+		assert.True(t, found)
+		assert.Equal(t, value, data)
+	})
+	t.Run("check temp batch is not nil", func(t *testing.T) {
```
Comment (AdoAdoAdo) [inline trie/trieBatchManager/trieBatchManager_test.go:96]: I don't think the description is correct

File: trie/trieChangesBatch/trieChangesBatch_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	tcb := NewTrieChangesBatch()
+	tcb.insertedData[string(keyForDeletion)] = core.TrieData{
+		Key:     []byte("trieKey"),
+		Value:   []byte("trieValue"),
+		Version: core.NotSpecified,
+	}
+
+	tcb.MarkForRemoval(keyForDeletion)
+	assert.Equal(t, 0, len(tcb.insertedData))
+	assert.Equal(t, 1, len(tcb.deletedKeys))
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply trie/trieChangesBatch/trieChangesBatch_test.go:53]: Done

File: trie/trieBatchManager/trieBatchManager_test.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply trie/trieBatchManager/trieBatchManager_test.go:45]: Fixed

File: trie/trieBatchManager/trieBatchManager_test.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply trie/trieBatchManager/trieBatchManager_test.go]: Indeed, changed

**Review by AdoAdoAdo — DISMISSED** [review]

File: trie/trieBatchManager/trieBatchManager_test.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	assert.False(t, tbm.isUpdateInProgress)
+	assert.False(t, check.IfNil(tbm.currentBatch))
+	assert.True(t, check.IfNil(tbm.tempBatch))
+}
+
+func TestTrieBatchManager_TrieUpdateInProgress(t *testing.T) {
+	t.Parallel()
+
+	tbm := NewTrieBatchManager()
+	tbm.currentBatch.Add([]byte("key1"), core.TrieData{})
```
Comment (ssd04) [inline trie/trieBatchManager/trieBatchManager_test.go:26]:
```suggestion
	tbm.Add([]byte("key1"), core.TrieData{})
```
use Add method directly? and in all other places

File: trie/trieBatchManager/trieBatchManager_test.go
Comment (ssd04) [inline trie/trieBatchManager/trieBatchManager_test.go:1]: should we consider a concurrency unit test?

File: trie/trieBatchManager/trieBatchManager_test.go (hunk 4 above)
Comment (BeniaminDrasovean) [reply trie/trieBatchManager/trieBatchManager_test.go:26]: Done.

File: trie/trieBatchManager/trieBatchManager_test.go
Comment (BeniaminDrasovean) [reply trie/trieBatchManager/trieBatchManager_test.go:1]: This will come in the next PRs in which I will also refactor some mutexes inside the trie

**Review by ssd04 — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**